src/algorithm/frames.hxx

```diff
@@ -138,7 +134,13 @@ namespace pinocchio
       {
         J_.col(j) = oMframe.actInv(Motion(data.J.col(j))).toVector(); // TODO: use MotionRef
       }
-      return;
+
+      if (rf == LOCAL_CARTESIAN_ORIENTED) {
+        Matrix6xLike J_tmp;
+        J_tmp.resize(6, model.nv);
+        J_tmp = SE3(data.oMf[frame_id].rotation(), Eigen::Vector3d::Zero()).toActionMatrix() * J;
```

Comment:
`Eigen::Vector3d` should not be employed directly. Pinocchio is templated by scalar type. The same holds true for `SE3`. Appropriate typedefs parametrized bu `Scalar` are to be preferred.

Reply:
I was not thinking about the parametrized by `Scalar` thing. Thanks for pointing this out.